### ISSUE TITLE
Time offset record

### ISSUE THREAD

**Issue by gmaclennan** [digidem/mapeo-core#55]
During our recent workshop in N. Peru we were initially flummoxed when the datetime filters did not work, and then realised that all the points had been recorded in 2017 because the phones were setup offline so their clocks never synced and we forgot to set them. I think this can be very common in the field - also some phones when left uncharged for a few days can forget the time and reset to 2017 or something.

I tested and confirmed that with https://github.com/ivanstan/react-native-nmea-library and it's possible to read the (correct) time from the GPS (Android only - iOS does not expose raw GPS data). We discussed setting the device clock based on the GPS time, but it looks like Android apps can't have permission to do this (makes sense), so we can only prompt the user to correct the time, but can't enforce it.

I think the best solution might be writing the time offset somewhere, either as a separate record to the hypercore, or maybe (better, I think) writing it as a property on individual records. Because of the delay (we can't guarantee the offset is accurate, but it should be correct to within a few seconds) I think this is more of a guidance than a source of truth, and probably the device time should be trusted if the offset is small. Therefore maybe this should be `observation.metadata.timeOffset` rather than a top-level `observation.timeOffset`, and when it is greater than a certain value, the frontend can calculate the created_at taking into consideration the offset, and maybe also expose that to the user.

Thoughts @noffle @aliya-ryan ?

**Comment by aliya-ryan** [comment]
Yes I think what you suggest sounds good, if I understand it all correctly. That is, if there is a big difference between phone recorded time and GPS time (more than a minute), then a message could come up to the user to ask them to check the phone time and date. And also for the GPS time to be recorded in the data.

Aliya Ryan | Digital Democracy  | UK: +44 (0)7518 454349 | aliya@digital-democracy.org
